Fix evohome HVAC modes for VisionPro Wifi systems

zxdavb opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR fixes a bug where some (i.e. a very small % of evohome) systems cannot set the HVAC mode and includes tests that confirm the bug is fixed.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

VisionPro Wifi systems are compatible with Honeywell (now Resideo) TCC, but have slightly different operating modes (e.g. Off instead of HeatingOff). This was mentioned in #32585 (which was closed when the primary issue was addressed in #32604).

Thus, these specific systems cannot be turned on/off via the UI or with the usual service calls.


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

zxdavb committed:
fix handling of edge case

zxdavb committed:
add tests of controller HVAC modes

zxdavb committed:
rename attr, add test tweak

zxdavb committed:
rename symbol for clarity

zxdavb committed:
no need to freeze time (yet)

zxdavb committed:
remove if from test, tweak

zxdavb committed:
tidy up

zxdavb committed:
Merge remote-tracking branch 'upstream/dev' into evo_fix_vizpro

zxdavb committed:
remove unneeded TYPE_CHECKING

oyvindwe commented on homeassistant/components/evohome/entity.py:
```diff
@@ -101,6 +100,8 @@ def __init__(
         """Initialize an evohome-compatible child entity (DHW, zone)."""
         super().__init__(evo_broker, evo_device)
 
+        self._evo_tcs = evo_device.tcs
```
Only used in `EvoZone` subclass, should it be set there instead? (I did check that `tcs` is available on both `evo.HotWater` and `evo.Zone`, as it previously was retrieved from `EvoBroker`.)

zxdavb replied:
DHW doesn't use the **tcs**, but it will do so in the future.

This PR moves that attr from all evohome entities (controller/TCS, DHW & zones), and leaves it only with child entities (DHW & zones), so is somewhat of an improvement...

I'd like to leave it in.

oyvindwe replied:
Sounds good to me!

zxdavb committed:
Merge remote-tracking branch 'upstream/dev' into evo_fix_vizpro

cdce8p committed:
Update eq3btsmart to 1.2.0 (#128808)

allenporter committed:
Bump google-nest-sdm to 6.1.0 (#128812)

epenet committed:
Use new reauth helpers in weatherflow_cloud (#128821)

epenet committed:
Use new reauth helpers in wallbox (#128820)

balloob committed:
Simplify custom component loading (#128813)

joostlek committed:
Bump spotifyaio to 0.7.1 (#128807)

epenet committed:
Use new reauth helpers in webostv (#128823)

epenet committed:
Use new reauth helpers in whirlpool (#128825)

tronikos committed:
Handle invalid zeroconf messages in Android TV Remote (#128819)

epenet committed:
Use new reauth helpers in xiaomi_ble (#128827)

epenet committed:
Use new reauth helpers in yale (#128828)

cdce8p committed:
Update zhong-hong-hvac to 1.0.13 (#128822)

epenet committed:
Use new reauth helpers in vicare (#128779)

frenck committed:
Auto lower case username for Schlage auth flows (#128730)

joostlek commented on tests/components/evohome/test_services.py:
This is testing services defined in `climate.py` if I understood correctly. We should test that in `test_climate.py`

Same comment as other PR, we should not touch the internals to test them

zxdavb replied:
> This is testing services defined in **climate.py** if I understood correctly. We should test that in **test_climate.py**

_(I note this code needs refactoring but, to address your concern...)_

They are a) domain-specific services, `hass.services.async_services_for_domain(DOMAIN)`. 

Not every evohome system supports the same service calls, so they are registered dynamically in **\_\_init__.py** in `setup_service_functions()`.

For those two reasons, I put them in **test_services.py**...

NOTE: Some do not apply to **Climate**/**WaterHeater** entities at all, and _should not_ be in **test_climate.py**/**test_water_heater.py**, e.g.:
```python
hass.services.async_register(DOMAIN, EvoService.REFRESH_SYSTEM, force_refresh)
```
... and (to a lesser extent):
```python
hass.services.async_register(DOMAIN, EvoService.RESET_SYSTEM, set_system_mode)
```

SHoudl I put them in **test_init.py**, or can I keep **test_services.py** (much like **test_storage.py**)?

bouwew committed:
Bump plugwise to v1.4.3 (#128773)

epenet committed:
Use new reauth helpers in weheat (#128824)

epenet committed:
Use new reauth helpers in youtube (#128835)

epenet committed:
Use new reauth helpers in yolink (#128834)

cdce8p committed:
Update attrs to 24.2.0 (#126656)

joostlek committed:
Add Spotify to strict typing (#128846)

epenet committed:
Use new reauth helpers in yalexs_ble (#128831)

epenet committed:
Use new reauth helpers in withings (#128826)

peteS-UK committed:
Add New Music Category for Media Browser (#128147)

LunaBytesBack committed:
Add Twitch stream viewer as readable data for integration (#128787)

CFenner committed:
Align consumption sensor names in ViCare integration (#127888)

oliverwoodings committed:
Reduce the size of the Nest event media storage cache (#128855)

Reduce max media items per nest device

alexkn committed:
Add humidity to KNX climate (#128844)

epenet committed:
Use new reauth helpers in yale_smart_alarm (#128836)

allenporter committed:
Bump google-nest-sdm to 6.1.3 (#128871)

Danielhiversen committed:
Bump pyTibber to 0.30.3 (#128860)

Xitee1 committed:
Remove myself from roomba codeowners (#128858)

bdraco committed:
Bump habluetooth to 3.6.0 (#128815)

joostlek committed:
Add audio feature sensors to Spotify (#128785)

emontnemery committed:
Improve entity cached attributes (#128876)

miaucl committed:
Use runtime_data for Swiss Public Transport (#128369)

* use runtime_data instead of hass.data[<key>]

* fix service response export type

* reduce runtime_data to be just the coordinator

* fix rebase

* fix ruff

* address reviews

* address reviews

* no general core import

* no general config_entries import

* fix also for services

* remove untyped config entry

* remove unneeded cast

piitaya committed:
Add translations for Netatmo thermostat preset modes (#128890)

miaucl committed:
Simplify Swiss public transport coordinator (#128891)

MindFreeze committed:
Include Z-Wave JS lowSecurityReason in node added websocket message (#128896)

* Propagate lowSecurityReason to FE when adding a zwavejs device insecurely

* update tests

emontnemery committed:
Remove dead code from concord232 (#128907)

sdb9696 committed:
Add reconfigure flow to ring integration (#128357)

Co-authored-by: Christopher Fenner <9592452+CFenner@users.noreply.github.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

emontnemery committed:
Add update_percentage state attribute to update entity (#128877)

* Add update_percentage state attribute to update entity

* Update tests

* Update tests

epenet committed:
Push real binary sensor states to state machine in tests (#128894)

epenet committed:
Use STATE_ON/STATE_OFF constants in template test (#128883)

10100011 committed:
Bump pyopenweathermap to v0.2.1 (#128892)

myztillx committed:
Add ecobee set_sensors_used_in_climate service (#102871)

* Add set_active_sensors Service

* Remove version bump from service addition commit

* Reviewer suggested changes

* Changed naming to be more clear of functionality

* Adjusted additional naming to follow new convention

* Updated to pass failing CI tests

* Fix typo

* Fix to pass CI

* Changed argument from climate_name to preset_mode and changed service error

* Made loop more clear and changed raised error to log msg

* Fix typo

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* Removed code that was accidentally added back in and fixed mypy errors

* Add icon for service

* Added sensors as attributes and updated tests

* Revert changes made in #126587

* Added tests for remote_sensors and set_sensors_used_in_climate

* Changed back to load multiplatforms (#126587)

* Check for empty sensor list and negative tests for errors raised

* Added tests and fixed errors

* Add hass to class init to allow for device_registry lookup at startup and check for name changed by user

* Added tests to test the new functions

* Simplified code and fixed testing error for simplification

* Added freeze in test

* Fixed device filtering

* Simplified code section

* Maintains the ability to call `set_sensors_used_in_climate` function even is the user changes the device name from the ecobee app or thermostat without needing to reload home assistant.

* Update tests with new functionality. Changed thermostat identifier to a string, since that is what is provided via the ecobee api

* Changed function parameter

* Search for specific ecobee identifier

* Moved errors to strings.json

* Added test for sensor not on thermostat

* Improved tests and updated device check

* Added attributes to _unrecoreded_attributes

* Changed name to be more clear

* Improve error message and add test for added property

* Renamed variables for clarity

* Added device_id to available_sensors to make it easier on user to find it

---------

Co-authored-by: Robert Resch <robert@resch.dev>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

Noltari committed:
Add Airzone switch entities to zones (#124562)

Noltari committed:
Add new QNAP QSW uptime timestamp sensor (#122589)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

emontnemery committed:
Remove explicit templating of persistent_notification service data (#128903)

mdegat01 committed:
Remaining addon management to aiohasupervisor (#128484)

* Move set addon options to aiohasupervisor

* addon stats to aiohasupervisor and test fixes

* addon changelogs to aiohasupervisor

* Raise correct error for library in tests

* Cache client in instance property

* Use singleton method rather then HassIO instance method

* Mock supervisor client in more tests

Noltari committed:
Add Airzone Cloud main zone mode select (#125918)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

Noltari committed:
Add Airzone Cloud switch entities to zones (#125917)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

DurandAN committed:
Add SIA alarm code (#127467)

barryvdh committed:
Add floor heating device valve positions in Homematic IP Cloud  (#122759)

* Update sensor.py for new FALMOT Sensors

First Integration attemp to support ValvePosition as Sensor for HmIP-FALMOT-C12

* Update sensor.py

* Update sensor.py

* Add Valve Position to FALMOT-C12

* modified: devcontainer

* Service für minimum vale postion hinzugefügt.

* update to services

* Service call optimized

* Add valvePosition to HomematicIP Cloud for Falmot-C12 and show only channels that are connected with an motorized actuator

* Fix some tests

* Add icon for service

* Fix tests, add check for ValveState in icon

* Remove minimum valve service

* REmove minimum valve

* Use list comprehension for devices, support other terminal blocks

* Remove unused constant

* Check correct channel

---------

Co-authored-by: thecem <46648579+thecem@users.noreply.github.com>

tr4nt0r committed:
Add fan `set_speed` support for Xiaomi Mi Air Purifier 3C (#126870)

gjohansson-ST committed:
Add config flow to local_file (#125835)

* Add config flow to local_file

* Small mods

* Add/fix tests

* Fix

* slug

* Fix strings

* Mod strings

tr4nt0r committed:
Add firmware update entity to IronOS integration (#123031)

chemelli74 committed:
Add diagnostics to Comelit SimpleHome (#128794)

* Add diagnostics to Comelit SimpleHome

* add test

* add missing tests

* introduce SnapshotAssertion

* cleanup

* exclude date based props

emontnemery committed:
Deprecate entity_id template variable in camera services (#128592)

* Deprecate entity_id template variable in camera services

* Update snapshots

* Tiny lang tweak

* Fix translation

---------

Co-authored-by: Franck Nijhof <git@frenck.dev>

chemelli74 committed:
Add diagnostics to Vodafone Station (#128923)

* Add diagnostics to Vodafone Station

* cleanup and exclude props based on date

emontnemery committed:
Add update_percentage property to update entity (#128908)

gjohansson-ST committed:
Allow Trend title to be translated (#128926)

gjohansson-ST committed:
Fix description placeholder in fibaro reauth (#128925)

gjohansson-ST committed:
Allow Random title to be translated (#128928)

gjohansson-ST committed:
Bump holidays to 0.59 (#128924)

emontnemery committed:
Remove explicit templating of telegram_bot service data (#128906)

emontnemery committed:
Remove explicit templating of minio service data (#128905)

emontnemery committed:
Remove explicit templating of velbus service data (#128904)

emontnemery committed:
Remove explicit templating of logbook service data (#128902)

gjohansson-ST committed:
Allow Timer title to be translated (#128927)

rappenze committed:
Fix description placeholder in brunt reauth (#128933)

* Fix description placeholder in brunt reauth

* Update homeassistant/components/brunt/config_flow.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

* Update homeassistant/components/brunt/config_flow.py

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

---------

Co-authored-by: Jan-Philipp Benecke <github@bnck.me>

qwell committed:
Add subscription tier attribute to Twitch integration. (#128870)

* Add subscription tier to Twitch integration.

* Add test for Twitch tiers.  Tests do not currently pass, so this is only theoretical.

* Fix variable type

* Show tier levels as 1,2,3 instead of the raw API values of 1000,2000,3000.

* Make Twitch subscription tier fixtures strings.

* Use proper assertion value for subscription tier test.

Edited on a bus on my phone. 😎

* Update homeassistant/components/twitch/coordinator.py

* Update tests/components/twitch/test_sensor.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

rahulsamant37 committed:
Add missing strings for mold indicator (#128205)

* Add missing localization keys for random component configuration

* Add missing localization keys for mold_indicator component configuration

* one_integration_at_a_time

* Fix localization strings for mold_indicator: use direct values instead of non-existing keys

* Fix localization strings for mold_indicator: use direct values instead of non-existing key

* Add missing translations for Mold Indicator helper

* correcting it for hassfest

* Fixes

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

gjohansson-ST committed:
Drop not needed reauth strings in tplink (#128937)

epenet committed:
Use new reauth helpers in unifi (#128837)

* Use new reauth helpers in unifi

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

* Update config_flow.py

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

rappenze committed:
Fix description placeholder in imap reauth (#128940)

gjohansson-ST committed:
Implement new state property for alarm_control_panel which is using an enum (#126283)

* Alarm state from enum

* Fixes

* Set final

* Fix rebase

* Test const

* Fix breaking version

* Fix other for alarm_control_panel

* Fix integrations

* More

* More

* More

* More

* Fix zha

* Replace _attr_state

* Fix alarm_control_panel

* Fix tests

* Fixes

* Mods

* Change some

* More

* More

* More

* Tests

* Last tests

* Return enum

* Fix zha

* Remove not needed check

* Fix wording

* Fix homekit

* Mod prometheus

* Fix mypy

* Fix homekit

* Fix ifttt

rappenze committed:
Fix description placeholder in transmission reauth (#128938)

rytilahti committed:
Add motion detected binary_sensor for tplink (#127883)

* Add motion binary_sensor for tplink

* Remove strings definition as we have device class that handles this

* Simplify instructions

* Remove mentions about fixture creation and snapshot updates as requested

* re-add newline

rytilahti committed:
Expose tplink temperature sensor as measurement (#128640)

Add state_class=measurement to the temperature sensor, making it available for long-term statistics.

gjohansson-ST committed:
Fix flaky update coordinator test (#128943)

bdraco committed:
Bump PySwitchbot to 0.49.0 (#128945)

bdraco committed:
Bump xiaomi-ble to 0.33.0 (#128946)

cdce8p committed:
Update astroid to 3.3.5 (#128948)

bdraco committed:
Bump yarl to 1.16.0 (#128941)

allenporter committed:
Bump gcal_sync to 6.2.0 (#128949)

bdraco committed:
Bump orjson to 3.10.9 (#128952)

bdraco committed:
Bump pySwitchbot to 0.50.1 (#128953)

changelog: https://github.com/Danielhiversen/pySwitchbot/compare/0.49.0...0.50.1

NickM-27 committed:
Add snapshot service to image entity (#110057)

* Add service definition for saving snapshot of image entity

* Add service to image

* Add tests for image entity service

* Fix tests

* Formatting

* Add service icon

* Formatting

* Formatting

* Raise home assistant error instead of single log error

* Correctly pass entity id

* Raise exception from existing exception

* Expect home assistant error

* Fix services example

* Add test for templated snapshot

* Correct icon service config

* Set correct type for service template

* Remove unneeded

Co-authored-by: Erik Montnemery <erik@montnemery.com>

* remove template

* fix imports

* Update homeassistant/components/image/__init__.py

* Apply suggestions from code review

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

osohotwateriot committed:
Add OSO Energy services (#118770)

* Add OSO Energy services

* Fixes after review

* Add tests for OSO Energy water heater

* Fixes after review

* Revert changes for service schema in OSO Energy

* Improve osoenergy unit tests

klejejs committed:
Change Stun server port to 80 (#128879)

allenporter committed:
Fix google tasks todo docstrings (#128978)

bdraco committed:
Add support for fetching bindkey from Mi cloud (#128394)

cdce8p committed:
Fix zha test RuntimeWarnings (#128975)

chemelli74 committed:
Bump aiocomelit to 0.9.1 (#128977)

* Bump aiocomelit to 0.9.1

* remove exception

chemelli74 committed:
Bump aiovodafone to 0.6.1 (#128976)

* Bump aiovodafone to 0.6.1

* remove exception

bdraco committed:
Bump PySwitchBot to 0.51.0 (#128990)

Petro31 committed:
Add limited template to at field for time triggers (#126584)

* Add limited template to at field for time triggers

* fix mypy

* Fix comments

* fix-tests

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Noltari committed:
Update aioairzone-cloud to v0.6.8 (#128992)

swoga committed:
Bump axis to v63 (#129005)

Lash-L committed:
Bump python-roborock to 2.6.1 (#128804)

lektribot committed:
Bump lektricowifi to 0.0.43 (#128979)

jpbede committed:
Use ConfigEntry.runtime_data in gardena_bluetooth (#129000)

epenet committed:
Improve template docstring (#128967)

gjohansson-ST committed:
Fix step in presets for generic thermostat (#128922)

Shulyaka committed:
Expose scripts with no fields as entities (#123061)

pvizeli committed:
Fix FUNDING.yml to OHF (#129013)

wendevlin committed:
Add Hassio HTTP logs/follow to allowed paths (#126606)

* Add logs/follow to admin paths in hassio.http

* Add tests for logs/follow admin paths in hassio.http

* Add tests for logs/follow admin paths in hassio.http

* Add compress and timeout exclusions for hassio http api

* Fix should_compress usage in hassio/ingress

* Add missing follow exceptions for hassio/http

* Add hassio range header forward for logs endpoints

* Fix test syntax hassio/http

bdraco committed:
Bump orjson to 3.10.10 (#129015)

changelog: https://github.com/ijl/orjson/compare/3.10.9...3.10.10

crug80 committed:
Adjust logging level in ModBus (#128980)

Fix issue 127570 in ModBus Component

zweckj committed:
Remove battery device class from bmw secondary sensor (#128970)

Remove battery device class

emontnemery committed:
Refactor camera.webrtc.register_ice_server (#129024)

* Refactor camera.webrtc.register_ice_server

* Apply suggestions from code review

Co-authored-by: Robert Resch <robert@resch.dev>

* Add missing import

---------

Co-authored-by: Robert Resch <robert@resch.dev>

edenhaus committed:
Add go2rtc binary config to expose api only on localhost (#129025)

dependabot[bot] committed:
Bump github/codeql-action from 3.26.13 to 3.27.0 (#129019)

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 3.26.13 to 3.27.0.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/v3.26.13...v3.27.0)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kingy444 committed:
Powerview migrate scene to string unique_id (#128131)

cereal2nd committed:
Bump pyduotecno to 2024.10.1 (#128968)

liudger committed:
Bump python bsblan version 0.6.4 (#128999)

emontnemery committed:
Allow configuring WebRTC stun and turn servers (#128984)

* Allow configuring WebRTC stun and turn servers

* Add tests

* Remove class WebRTCCoreConfiguration

unfug-at-github committed:
Have statistics functions return a meaningful, non-none result even if only one value is available (#127305)

* have statistics functions return a meaningful, non-none result even if only one value is available

* improved code coverage

lektribot committed:
Add switch platform to the Lektrico integration (#126721)

Shutgun committed:
Fix devolo_home_network devices not reporting a MAC address (#129021)

dependabot[bot] committed:
Bump actions/cache from 4.1.1 to 4.1.2 (#129018)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

qwell committed:
Remove deprecated channel views attribute from Twitch (#129008)

epenet committed:
Use runtime_data in balboa (#129035)

jbergler committed:
Add `completed` to the wait variable when using triggers (`wait_for_trigger`) (#123427)

* Add support for the wait.completed variable when using wait with triggers

* Remove junk comment

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>

epenet committed:
Use runtime_data in bang_olufsen (#129037)

kbickar committed:
Create tests for sense integration (#128418)

* Create tests for sense integration

* Rearrange files

* Update to use snapshots

* Update tests/components/sense/__init__.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/__init__.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/test_binary_sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Update tests/components/sense/test_sensor.py

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

* Add missing imports

---------

Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

joostlek committed:
Bump yt-dlp to 2024.10.22 (#129034)

bdraco committed:
Bump sensorpush-ble to 1.7.0 (#128951)

changelog: https://github.com/Bluetooth-Devices/sensorpush-ble/compare/v1.6.2...v1.7.0

gjohansson-ST committed:
Fix calculation of attributes in group sensor (#128601)

* Fix calculation of attributes in group sensor

* Fixes

* Fixes

* Make module level function

Thomas55555 committed:
Fix get_time_zone annotations in dt_util (#129050)

bdraco committed:
Fix cancellation leaking upward from the timeout util (#129003)

mxr committed:
Add 'select' to configure Schlage locks "Auto Lock Time" (#123758)

dependabot[bot] committed:
Bump actions/checkout from 4.2.1 to 4.2.2 (#129063)

Xeio committed:
Use fan mode when heat/cool is idle in homekit_controller (#128618)

bramkragten committed:
Update frontend to 20241002.4 (#129049)

epenet committed:
Remove bloomsky integration (#129073)

* Small refactor to bloomsky

* Remove bloomsky integration

* Update integrations.json

jb101010-2 committed:
Switch from pysuez to pysuezV2 in Suez Water (#127113)

Nebula83 committed:
Add config flow to Onkyo (#117319)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Artur Pragacz <49985303+arturpragacz@users.noreply.github.com>
Co-authored-by: Artur Pragacz <artur@pragacz.com>
Co-authored-by: Joostlek <joostlek@outlook.com>

joostlek committed:
Add config flow to Smarty (#127540)

Co-authored-by: Sid <27780930+autinerd@users.noreply.github.com>

emontnemery committed:
Move core config functionality to its own module (#129065)

* Move core config functionality to its own module

* Adjust test

epenet committed:
Use runtime_data in blebox (#129070)

epenet committed:
Use runtime_data in blink (#129072)

joostlek committed:
Add unique id and tests for Smarty (#129078)

LouisChrist committed:
Add tests for media_player to bluesound integration (#125864)

epenet committed:
Use runtime_data in bluemaestro (#129085)

epenet committed:
Use runtime_data in blue_current (#129084)

epenet committed:
Use runtime_data in buienradar (#129087)

cdce8p committed:
Update pytest warnings filter (#129075)

zxdavb committed:
Switch to using a fixture for evohome WaterHeater tests (#127701)

Co-authored-by: Christopher Fenner <9592452+CFenner@users.noreply.github.com>
Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Erik Montnemery <erik@montnemery.com>
Co-authored-by: Robert Resch <robert@resch.dev>
Co-authored-by: thecem <46648579+thecem@users.noreply.github.com>
Co-authored-by: Franck Nijhof <git@frenck.dev>
Co-authored-by: Jan-Philipp Benecke <github@bnck.me>
Co-authored-by: G Johansson <goran.johansson@shiftit.se>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: epenet <6771947+epenet@users.noreply.github.com>

epenet committed:
Use shorthand attribute in template binary sensor (#128966)

AlD committed:
Remove DHCP match from awair (#129047)

Co-authored-by: Joostlek <joostlek@outlook.com>

noahhusby committed:
Fix update callback in Cambridge Audio test (#129092)

bdraco committed:
Add Meter Pro support to SwitchBot (#128991)

epenet committed:
Use runtime_data in bsblan (#129089)

qwell committed:
Reduce the number of API calls in Twitch integration (#128996)

tr4nt0r committed:
Add buttons to cast skills in Habitica integration (#126350)

mkmer committed:
Handle temprorary hold in Honeywell (#128460)

Shutgun committed:
Fix adding multiple devices simultaneously to devolo Home Network's device tracker (#129082)

autinerd committed:
Simplify webmin tests to use snapshot_platform (#127754)

noahhusby committed:
Add play media capability to Cambridge Audio (#129002)

autinerd committed:
Bump ruff to 0.7.1 (#129102)

emontnemery committed:
Allow update entities to report progress as a float (#128930)

* Allow update entities to report progress as a float

* Add test

* Update snapshots

* Update recorder test

* Use _attr_* in MockUpdateEntity

Thomas55555 committed:
Bump aioautomower to 2024.10.3 (#128788)

joostlek committed:
Add coordinator to Smarty (#129083)

* Add coordinator to Smarty

* Add coordinator to Smarty

* Fix

mdegat01 committed:
Fix null hass error in supervisor update entities (#129030)

* Fix null hass error in supervisor update entities

* Share the supervisor client with coordinator

* Remove unnecessary patch of helper

* Attribute not property

joostlek committed:
Add entity descriptions to Smarty Binary sensor (#129110)

dontinelli committed:
Add missing state_class to sensors in solarlog (#128296)

* Add missing state_class

* Update snapshot

edenhaus committed:
Use webrtc-models package (#129032)

kbickar committed:
Add typing for sense component (#129119)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

dependabot[bot] committed:
Bump actions/setup-python from 5.2.0 to 5.3.0 (#129121)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 5.2.0 to 5.3.0.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v5.2.0...v5.3.0)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tr4nt0r committed:
Set up single coordinator for all config entries in IronOS (#129108)

zxdavb committed:
Switch to using a fixture for evohome Climate tests (of zones) (#129100)

tr4nt0r committed:
Add entity picture for healing potion in Habitica (#129107)

tr4nt0r committed:
Refactor services setup in Habitica integration (#128186)

mback2k committed:
Prevent duplicate WMS WebControl pro config entry creation (#128315)

MindFreeze committed:
Resume adding Z-Wave device if the page is refreshed (#129081)

* ZwaveJS: Resume adding a device if the page is refreshed

* add test

* address PR comments

noahhusby committed:
Add additional tests to Cambridge Audio (#128213)

zxdavb commented:
Easiest to close this PR and start again.
